### PR TITLE
build!: drop node 10 from engines list, update typescript to 4.6.3

### PR DESCRIPTION
**Pull request by ddelgrosso1**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com//issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node: [10, 12, 14, 16]
+        node: [12, 14, 16]
```
Comment (ddelgrosso1) [inline .github/workflows/ci.yaml:12]: OwlBot seems upset that I touched this. Should I have left this alone?

**Comment by sofisl** [comment]
@ddelgrosso1, everything looks good here except that the post-processor doesn't seem to be picking up the new ci.yaml file. It should look like: `node: [12, 14, 16]` and the SRS file should be testing 12, 14, and 16, since we're dropping Node 10 now. I wonder if the force-push perhaps overwrote what the post-processor should have done,  would you mind attempting to re-push to see if the postprocessor runs? You can also attempt to run the script that I wrote in a new branch if that doesn't work, should be the same changes you've already made.

**Comment by sofisl** [comment]
The other question I have is for @bcoe, should these be build! or feat!

**Comment by ddelgrosso1** [comment]
@sofisl the reason the post processor didn't run is because I work from my own fork. I always have to manually trigger it. I've updated the 2 .yml files you called  out.

**Review by sofisl — APPROVED** [review]

**Review by bcoe — APPROVED** [review]

**Pull request merged**